state/selectors/measurementPeriodString.ts

```diff
@@ -0,0 +1,35 @@
+import { selector } from 'recoil';
+import { measurementPeriodState } from '../atoms/measurementPeriod';
+
+export const measurementPeriodStringState = selector<string>({
```

Comment:
This is a nitpick, but I feel like, conceptually, this doesn't really need to be a selector. It's just a string that is rendered based on the values of another atom, however the rest of the app doesn't need access to it. It just renders once at the top. It might make sense to move the date string logic into the actual component that renders the banner. Your call though